addons/barcodes_gs1_nomenclature/models/barcode_rule.py

```diff
@@ -0,0 +1,49 @@
+import re
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class BarcodeRule(models.Model):
+    _inherit = 'barcode.rule'
+
+    encoding = fields.Selection(
+        selection_add=[('gs1-128', 'GS1-128')],
+        ondelete={'gs1-128': 'set default'})
+    type = fields.Selection(
+        selection_add=[
+            ('location_dest', 'Destination location'),
+            ('qty_done', 'Quantity'),
+            ('use_date', 'Best before Date'),
+            ('expiration_date', 'Expiration Date'),
+            ('packaging_date', 'Packaging Date')
+        ], ondelete={
+            'location_dest': 'set default',
+            'qty_done': 'set default',
+            'use_date': 'set default',
+            'expiration_date': 'set default',
+            'packaging_date': 'set default',
+        })
+    is_gs1_nomenclature = fields.Boolean(related="barcode_nomenclature_id.is_gs1_nomenclature")
+    gs1_content_type = fields.Selection([
+        ('date', 'Date'),  # To able to transform YYMMDD date into Odoo datetime
+        ('measure', 'Measure'),  # UOM related
+        ('identifier', 'Numeric Identifier'),  # When there is a numeric fixed length where there is a check digit ()
+        ('alpha', 'Alpha-Numeric name'),
+    ], string="GS1 Content Type")
+    gs1_decimal_usage = fields.Boolean('Decimal', help="If True, use the last digit of AI to dertermine where the first decimal is")
+    associated_uom_id = fields.Many2one('uom.uom')
+
+    def _default_encoding(self):
```

Comment:
What it that ?

Reply:
It's the default encoding method I forgot to set on the field :sweat: